sw/device/silicon_creator/lib/drivers/keymgr.h

```diff
@@ -0,0 +1,61 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef OPENTITAN_SW_DEVICE_SILICON_CREATOR_LIB_DRIVERS_KEYMGR_H_
+#define OPENTITAN_SW_DEVICE_SILICON_CREATOR_LIB_DRIVERS_KEYMGR_H_
+
+#include <stddef.h>
+#include <stdint.h>
+
+#include "sw/device/silicon_creator/lib/error.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * Initializes the key manager.
+ *
+ * Initializes the key manager `entropy_reseed_interval` and advances the state
+ * into initialized.
+ *
+ * The working status of the key manager must be set to reset before
+ * calling this function otherwise it will return `kErrorKeymgrInternal`.
+ *
+ * @param entropy_reseed_interval Number of key manager cycles before the
+ * entropy is reseeded.
+ * @return The result of the operation.
+ */
+rom_error_t keymgr_init(uint16_t entropy_reseed_interval);
+
+/**
+ * Advances the state of the key manager to Creator Root Key state.
+ *
+ * This operation is non-blocking to allow the software to continue with other
+ * operations while the key manager is advancing its state. The caller is
+ * responsible for calling the `keymgr_check_creator_state()` at a later time
+ * to ensure the advance transition completed without errors.
+ *
+ * @param binding_value Software binding value extracted from the ROM_EXT
+ * manifest.
+ * @param max_key_version Maximum key version extracted from the ROM_EXT
+ * manifest.
+ * @return The result of the operation.
+ */
+rom_error_t keymgr_advance_creator_state(const uint32_t binding_value[8],
```

Comment:
Array params aren't a good idea... the compiler won't size check them. Can you add a struct that wraps a `u32[8]`?

(This is a bit asinine but it's C's fault for being a bad language, not mine.)

Reply:
It is fine :). 

@alphan Is it ok if we wrap this in a struct in the manifest so that we don't need to copy data?

Reply:
Definitely.

Reply:
Are we planning to call this from the ROM with the value of this field taken from the validated ROM_EXT that we choose to boot?  Our [documentation](https://docs.opentitan.org/hw/ip/keymgr/doc/) is a bit inconsistent in explaining this:

The "Key Manager State" section doesn't mention a software binding value for the CreatorRootKey derivation, but does for the Owner{Intermediate,Root}Key derivations.  However, later in the document in the "Software Binding" section, we explain that a binding value is used while advancing to each of the CreatorRootKey, and Owner* states.

Reply:
I will add a separate header with the `binding_value` struct, so that we can integrate it with the manifest. 

As for the behavioral question from @cfrantz, the answer is yes. We are planning to configure the key manager inputs from validated data from the ROM_EXT manifest. I have added lowrisc/opentitan#6714 to continue this discussion, as there are additional considerations for the boot policy and shutdown modules that we need to close on.